src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,357 @@
+        dtype (numpy.dtype): the data type
+        metadata (dict str->val): The metadata
+        """
+        shape = (stack_info["dimension"]["x"], stack_info["dimension"]["y"])
+        tile_shape = (stack_info["mirrors"][0]["tile_width"], stack_info["mirrors"][0]["tile_height"])
+        # resolution is given in nanometers by catmaid.
+        resolution = {k: v * 1e-9 for k, v in stack_info["resolution"].items()}
+        metadata = {model.MD_PIXEL_SIZE: (resolution["x"], resolution["y"])}
+        dtype = numpy.uint8
+
+        maxzoom = stack_info.get("num_zoom_levels", -1)
+        if maxzoom == -1:
+            # if num_zoom_levels is -1 the maxzoom must be estimated by using: 2**maxzoom = max_dim / min_size
+            # Rewriting this results in maxzoom = log(max_dim / min_size) / log(2)
+            max_dim = max(shape)
+            min_size = min(tile_shape)
+            maxzoom = int(math.ceil(math.log(max_dim / min_size, 2)))
+
+        DataArrayShadow.__init__(self, shape, dtype, metadata, maxzoom=maxzoom, tile_shape=tile_shape)
+
+        self._base_url = base_url
+        self._session = requests.Session()
+        _, username, password = read_config_file(self._base_url, username=True, password=True)
+        self._auth = (username, password)
+        self._stack_info = stack_info
+        file_extension = self._stack_info["mirrors"][0]["file_extension"]
+        self._file_extension = file_extension[1:] if file_extension.startswith(".") else file_extension
+
+    def getTile(self, x, y, zoom, depth=0):
+        """
+        Fetches one tile
+        x (0<=int): X index of the tile.
+        y (0<=int): Y index of the tile
+        zoom (0<=int): zoom level to use. The total shape of the image is shape / 2**zoom.
+            The number of tiles available in an image is ceil((shape//zoom)/tile_shape)
+        depth (0<=int): The Z index of the stack.
+        return:
+            tile (DataArray): tile containing the image data and the relevant metadata.
+        """
+        tile_width, tile_height = self.tile_shape
+        tile_url = format_tile_url(
+            tile_source_type=self._stack_info["mirrors"][0]["tile_source_type"],
+            image_base=self._stack_info["mirrors"][0]["image_base"],
+            zoom=zoom,
+            depth=depth,
+            col=x,
+            row=y,
+            file_extension=self._file_extension,
+            tile_width=tile_width,
+            tile_height=tile_height,
+        )
+        try:
+            image = response_to_array(self._session.get(tile_url, auth=self._auth))
+        except HTTPError as e:
+            if e.response.status_code == 401:
+                raise AuthenticationError("Authentication failed while getting tiles at {}".format(tile_url))
+            else:
+                logging.error("No tile at %s (error %s), returning blank tile", tile_url, e.response.status_code)
+                image = numpy.zeros((tile_width, tile_height))
```

Comment:
You should return a numpy array of the same dtype as the usual ones -> `numpy.zeros(..., dtype=self.dtype)`.